addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxControlJalousie.java

```diff
@@ -216,24 +196,20 @@ public Double getPosition() {
     @Override
     public void onStateChange(LxControlState state) {
         // check position changes
-        if (state.getName().equals(STATE_POSITION) && targetPosition != null && targetPosition > 0
+        if (STATE_POSITION.equals(state.getName()) && targetPosition != null && targetPosition > 0
                 && targetPosition < 1) {
             // see in which direction jalousie is moving
-            LxControlState up = getState(STATE_UP);
-            LxControlState down = getState(STATE_DOWN);
-            if (up != null && down != null) {
-                Double currentPosition = state.getValue();
-                Double upValue = up.getValue();
-                Double downValue = down.getValue();
-                if (currentPosition != null && upValue != null && downValue != null) {
-                    if (((upValue == 1) && (currentPosition <= targetPosition))
-                            || ((downValue == 1) && (currentPosition >= targetPosition))) {
-                        targetPosition = null;
-                        try {
-                            stop();
-                        } catch (IOException e) {
-                            logger.debug("Error stopping jalousie when meeting target position.");
-                        }
+            Double currentPosition = state.getValue();
+            Double upValue = getStateValue(STATE_UP);
+            Double downValue = getStateValue(STATE_DOWN);
+            if (currentPosition != null && upValue != null && downValue != null) {
+                if (((upValue == 1) && (currentPosition <= targetPosition))
+                        || ((downValue == 1) && (currentPosition >= targetPosition))) {
+                    targetPosition = null;
+                    try {
+                        stop();
+                    } catch (IOException e) {
+                        logger.debug("Error stopping jalousie when meeting target position.");
```

Comment:
Is debug enough any other handling needed? Devices offline? Resetting things?

Reply:
If there is communication error, which is the only reason why this can fail, and blinds can't be stopped, so they can't, and will reach end position. The connection handling should figure it out and take care of reestablishment of the broken connection independently.

Reply:
👍